sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java

```diff
@@ -82,14 +83,27 @@ public ProviderResponse reinitPassword(
       ProviderResponse response =
           storeProvider
               .getWriterStore(realm, userStorage)
-              .reinitPassword(userId, password, pcr, providerRequest);
+              .reinitPassword(
+                  userId,
+                  password,
+                  changePasswordResetStatus,
+                  templateProperties,
+                  webserviceTag,
+                  providerRequest);
       sugoiEventPublisher.publishCustomEvent(
           realm,
           userStorage,
           SugoiEventTypeEnum.RESET_PASSWORD,
           Map.ofEntries(
-              Map.entry(EventKeysConfig.PASSWORD_CHANGE_REQUEST, pcr),
+              Map.entry(EventKeysConfig.PROPERTIES, templateProperties),
+              Map.entry(
+                  EventKeysConfig.MAIL,
+                  templateProperties.get("mail") != null
+                      ? templateProperties.get("mail")
+                      : (user.getMail() != null ? user.getMail() : "")),
```

Comment:
What happen if there is no mail in the template and in the object user and webserviceTag is mail ?

Reply:
Yes but we must be careful because the process for sending webhook is asynchronous, so it will be difficult for the webhook service to throw an exception for the user. But we need to be more flexible as possible at this level